### PR TITLE
[Reland] Add sym_size/stride/numel/storage_offset to native_function.…

### PR DESCRIPTION
**Pull request by SherlockNoMad**
…yaml (#91… (#91919)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/91919 Approved by: https://github.com/ezyang

Fixes #ISSUE_NUMBER

Pull Request resolved: https://github.com/pytorch/pytorch/pull/92402

Reviewed By: ezyang

Differential Revision: D42565586

Pulled By: SherlockNoMad

fbshipit-source-id: 1c2986e45307e076d239836a1b45441a9fa3c9d9
ghstack-source-id: 969f4928486e04c57aaf98e20e3c3ca946c51613

Fixes #ISSUE_NUMBER

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
I'm curious why you need these to be native function?

File: torch/_dynamo/eval_frame.py [hunk 1]
```diff
@@ -883,10 +883,15 @@ def __init__(
...
-            ]
+            self.new_args = []
+            for i in range(0, arg_len):
+                arg = super(ChangeInputOutputSignature, self).placeholder(f"arg{i}", (), {})
+                # Fill node.mata["val"] with faketensor from the input,
+                # if it's not found in matched_input_elements_positions
+                if i not in matched_input_elements_positions:
+                    arg.node.meta["val"] = fake_mode.from_tensor(flat_args[i])
+                self.new_args.append(arg)
+
```
Comment (SherlockNoMad) [inline torch/_dynamo/eval_frame.py:894]: Here!

Comment (tugsbayasgalan) [reply torch/_dynamo/eval_frame.py:894]: LGTM!

**Comment by SherlockNoMad** [comment]
> I'm curious why you need these to be native function?

They need to be properly aten op, so that they can be included in the Core Aten IR. 
Also, folks are expecting aten.sym_size to be an OpOverload https://github.com/pytorch/pytorch/issues/97201
Also, we need schema for them.

**Comment by albanD** [comment]
But all of these would work with a registration from python right?

**Comment by albanD** [comment]
Do you mean a libtorch-only build will need to access it?
The usual dispatcher API to get a kernel even if it wasn't defined in c++. We do that to get our python-registered triton kernel here for example: https://github.com/pytorch/pytorch/blob/8769fb854d816d223cfb513979e97e23a93bddee/aten/src/ATen/native/sparse/SparseBlasImpl.cpp#LL97C10-L97C14

**Review by zhxchen17 — APPROVED** [review]

**Comment by suo** [comment]
> Do you mean a libtorch-only build will need to access it? 

yeah

**Review by albanD — APPROVED** [review]
SGTM!

File: torch/_dynamo/eval_frame.py [hunk 2]
```diff
@@ -924,10 +924,21 @@ def __init__(
...
-            self.new_args = [
-                super(ChangeInputOutputSignature, self).placeholder(f"arg{i}", (), {})
-                for i in range(0, arg_len)
-            ]
+            self.new_args = []
+            for i in range(0, arg_len):
+                arg = super(ChangeInputOutputSignature, self).placeholder(
+                    f"arg{i}", (), {}
+                )
+                # Fill node.mata["val"] with faketensolintrunner from the input,
```
Comment (malfet) [inline torch/_dynamo/eval_frame.py:932]: @suo `lintrunner` is being immortalized in _dynamo comments :P

**Comment by kit1980** [comment]
FYI the correct linked diff for this PR is D45620400

**Pull request closed without merging**